pkg/bitcoin/electrum/electrum.go

```diff
@@ -771,9 +771,9 @@ func requestWithRetry[K interface{}](
 			requestCtx, requestCancel := context.WithTimeout(ctx, c.config.RequestTimeout)
 			defer requestCancel()
 
-			c.clientMutex.RLock()
```

Comment:
Nitpick: If we no longer use read locks, we can change the `clientMutex` type to `sync.Mutex`.

Reply:
Sure, [c0e9d25](https://github.com/keep-network/keep-core/pull/3701/commits/c0e9d25fc1868266d6afef35fe26253b05d8c9b2). Let's see what we agree on the discussion below. If we limit the scope of the fix, I'll revert this change. If we leave the scope wider, I'll leave it.